### ISSUE TITLE
Missing RDS Engine Types: db2-ae and db2-se

### ISSUE THREAD

**Issue by CajuCLC** [aws-cloudformation/cfn-lint#3138]
### CloudFormation Lint Version

cfn-lint 0.86.2

### What operating system are you using?

Windows

### Describe the bug

When using cfn-lint to validate AWS CloudFormation templates, the tool fails to recognize `db2-ae` and `db2-se` as valid RDS engine types, leading to incorrect validation errors.

```
cfn-lint --template .\rdsdb2.yml                         
W2031 You must specify a valid Default value for pEngine (db2-se). Valid values must match pattern Has to be one of [aurora, aurora-mysql, aurora-postgresql, mariadb, mysql, oracle-ee, oracle-ee-cdb, oracle-se2, oracle-se2-cdb, oracle-se1, oracle-se, postgres, sqlserver-ee, sqlserver-se, sqlserver-ex, sqlserver-web]
.\rdsdb2.yml:6:5

W2031 You must specify a valid allowed value for pEngine (db2-ae). Valid values must match pattern Has to be one of [aurora, aurora-mysql, aurora-postgresql, mariadb, mysql, oracle-ee, oracle-ee-cdb, oracle-se2, oracle-se2-cdb, oracle-se1, oracle-se, postgres, sqlserver-ee, sqlserver-se, sqlserver-ex, sqlserver-web]
.\rdsdb2.yml:10:9

W2031 You must specify a valid allowed value for pEngine (db2-se). Valid values must match pattern Has to be one of [aurora, aurora-mysql, aurora-postgresql, mariadb, mysql, oracle-ee, oracle-ee-cdb, oracle-se2, oracle-se2-cdb, oracle-se1, oracle-se, postgres, sqlserver-ee, sqlserver-se, sqlserver-ex, sqlserver-web]
.\rdsdb2.yml:11:9
```

### Expected behavior

cfn-lint should recognize `db2-ae` and `db2-se` as valid RDS engine types without error.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: RDS DB2 CloudFormation

Parameters:
  pEngine:
    Default: db2-se
    Description: The database engine
    Type: String
    AllowedValues:
      - db2-ae
      - db2-se

Resources:
  rDBInstance:
    Type: AWS::RDS::DBInstance
    Properties:
      Engine: !Ref pEngine
      ...
```

**Comment by CajuCLC** [comment]
I created the PR with changes. After making the changes on my local, no more errors:

```
cfn-lint --template .\rdsdb2.yml --format json
[]
```

**Comment by kddejong** [comment]
This has been released.
